src/simplewallet/simplewallet.cpp

```diff
@@ -362,6 +362,29 @@ bool simple_wallet::seed_set_language(const std::vector<std::string> &args/* = s
   return true;
 }
 
+bool simple_wallet::change_password(const std::vector<std::string> &args)
+{
+  // watch-only wallets don't need passwords, right?
+  if(m_wallet->watch_only())
+  {
+    fail_msg_writer() << tr("wallet is watch-only and has no seed");
+    return true;
+  }
+  
+  // prompts for a new password, this is not a new wallet so pass in false.
+  const auto pwd_container = tools::wallet2::password_prompt(false);
+  
+  if(!pwd_container)
+  {
+    fail_msg_writer() << tr("no password given");
+    return true;
```

Comment:
I think this should return false. Password-prompt doesn't fail if a user provides an empty password, it fails if `std::cin` cannot be read from or is not a tty. The semantics of returning a `boost::optional<>` here are misleading (thats on me), hopefully I can con people into letting `boost::expected` or similar into the codebase.